app/services/invites/create_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Invites
+  class CreateService < BaseService
+    def call(**args)
+      existing_invite = Invite.find_by(
+        organization_id: args[:organization_id],
+        email: args[:email],
+        status: :pending,
+      )
+
+      if existing_invite
+        return result.fail!(
+          code: 'invite_already_exists',
+          message: 'Invite already exists',
+        )
+      end
+
+      existing_membership = Membership.joins(:user).active.where(
```

Comment:
same here, validation should not be in the create scope but in a dedicated service!

Reply:
Done